### ISSUE TITLE
Grafana Cloud Loki data source: "log to trace" of the loki data source is broken for the new OTel logs format

### ISSUE THREAD

**Issue by cyrille-leclerc** [grafana/grafana#92699]
### What happened?

"log to trace" on the built-in Loki data source in Grafana Cloud is broken when adopting the nw OpenTelemetry logs format.

The reason is that the configured "Derived fields" of the built-in Loki data source expects th TraceID to be located in the log line (regex `[tT]race_?[iI][dD]"?[:=]"?(\w+)`) when the new Loki format for OpenTelemetry logs stores the TraceID in a metadata `trace_id`.

This bug is particularly painful as the builtin Loki data source configuration is readonly.

Actual

![Image](https://github.com/user-attachments/assets/b95f23f1-3544-4032-90d3-c4b9ebf20cff)

### What did you expect to happen?

Expected

![Image](https://github.com/user-attachments/assets/0fa02270-51c6-428f-a140-654ea89cd2cb)**

### Did this work before?

Never worked with the Loki V3 OTel logs format

### How do we reproduce it?

1. Send OpenTelemetry traces and logs to a Grafana Cloud instance
2. Explore the produced logs through Grafana Explore
3. see that the log 2 trace link is not displayed

### Environment (with versions)?

Grafana Cloud, August 2024

### Grafana platform?

I use Grafana Cloud

### Datasource(s)?

Loki and Tempo

**Comment by cyrille-leclerc** [comment]
Solution described on "[OpenTelemetry Collector Loki Exporter / Migration Instructions / Grafana visualizations migrations](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/lokiexporter/README.md#loki-data-source-log-to-trace)"

To enable the "logs to trace" navigation from Loki to Tempo, navigate to the Grafana Loki data source configuration screen, in the "Derived fields" section, update or create a derived field with:
* Name: `Trace ID`
* Type: `Label` (this `Label` name is missleading because it also supports Loki message metadata)
* Label: `trace_id`
* Internal link: activated
* Select the Tempo data source on which "trace to logs" is configured

Configuration screenshot:

![Image](https://github.com/user-attachments/assets/579a58bc-4fb2-4dcd-ade8-81c9fb0055f6)

**Comment by vlad-diachenko** [comment]
**The best fix**  would be to make Loki data uniform and extract `trace_id` on Loki side, if `trace_id` is not specified in the structured_metadata of the original write request. It would allow us to change default settings for "Derived fields" to extract trace ID from "Labels" by default. 
That said, it requires changes on Loki side.

**The easiest fix** is to add one more "Derived field" to extract trace ID from the "Labels" (it includes regular labels, parsed labels, and structured metadata). Like this:
![Image](https://github.com/user-attachments/assets/e869123c-42dc-4a34-9892-7e09a1141c86)
In this case, Grafana adds a link to the trace for both cases: for extracted trace ID from the logline and from structured metadata(OTEL case, but not only):
![Image](https://github.com/user-attachments/assets/c09f5939-a884-46f9-b814-1846f57786a4)
However, in this case, the user sees a few links, which can be confusing...

**Comment by cyrille-leclerc** [comment]
I like the solution with 2 derived fields because it looks very unlikely to me that a traceId would appear in the log body if it's stored as a metadata. @zeitlinger, @matt-hensley, or @maryliag WDYT of this solution?

**Comment by matt-hensley** [comment]
Agreed @cyrille-leclerc, it is unlikely the value would appear in both locations, and if it does this solution won't cause any problems. This looks like a good way to fix this quickly👍
